crates/cli/src/lib.rs

```diff
@@ -777,35 +812,44 @@ fn make_argv_envp<'a, I: IntoIterator<Item = S>, S: AsRef<OsStr>>(
 /// Run on the native OS (not on wasm)
 #[cfg(target_family = "unix")]
 fn roc_run_native<I: IntoIterator<Item = S>, S: AsRef<OsStr>>(
-    arena: Bump,
+    arena: &Bump,
     opt_level: OptLevel,
     args: I,
     binary_bytes: &[u8],
     expectations: VecMap<ModuleId, Expectations>,
     interns: Interns,
+    layout_interner: SingleThreadedInterner<Layout>,
 ) -> std::io::Result<i32> {
     use bumpalo::collections::CollectIn;
 
     unsafe {
         let executable = roc_run_executable_file_path(binary_bytes)?;
-        let (argv_cstrings, envp_cstrings) = make_argv_envp(&arena, &executable, args);
+        let (argv_cstrings, envp_cstrings) = make_argv_envp(arena, &executable, args);
 
         let argv: bumpalo::collections::Vec<*const c_char> = argv_cstrings
             .iter()
             .map(|s| s.as_ptr())
             .chain([std::ptr::null()])
-            .collect_in(&arena);
+            .collect_in(arena);
 
         let envp: bumpalo::collections::Vec<*const c_char> = envp_cstrings
             .iter()
             .map(|s| s.as_ptr())
             .chain([std::ptr::null()])
-            .collect_in(&arena);
+            .collect_in(arena);
+
+        let opt_level = OptLevel::Development;
```

Comment:
I think we should decouple the idea of "optimization" from the backend. We could add a `--backend=assembly|llvm|wasm` flag maybe? we will probably always want this option for development (e.g. have the llvm backend with the wasm target to use the llvm wasm backend without optimization). Thoughts?

Reply:
I like the decoupling idea. Maybe we can infer the backend based on the target and what optimization level you pass in, though - for example if default optimization level is development, we still probably want to prefer the dev backend, unless you overwrite the backend. I also think it could make sense for a `backend` flag only to be exposed in debug/nightly builds, so users can't change that.

Reply:
Yeah this seems like a risky thing to expose to all users, because then it ties our hands in terms of making future changes without breaking backwards compatibility.

For example, right now LLVM is an implementation detail to end users. We could switch to Cranelift, or [some future thing that's more awesome than LLVM], or whatever, and it wouldn't be a breaking API change because we don't offer you the ability to say "I want LLVM specifically."

That said, there are a number of things like this we've discovered which are useful to have for people developing Roc, but which shouldn't necessarily be exposed in public releases. Maybe it's time we made a system for doing what Ayaz just suggested - making certain options available when the compiler is built from source, but not when it's built for release.

Reply:
ok. Well for this PR I'll do the refactor to have a "backend" concept, we can decide how to expose it later.